### PR TITLE
Fixes #24351 - chart click through search

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Fixes long lasting regression of charts on dashboard not being clickable through.
Adds posibility to specify custom filters for chart segments.
Moves click handler from DonutChartService to ChartService, as there is no reason for
this funcionality not to be added to more chart types in the future.

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
Overseeds #8208 as it contains the fix.

File: webpack/assets/javascripts/react_app/components/ChartBox/ChartBox.js [hunk 1]
```diff
@@ -70,9 +66,7 @@ class ChartBox extends React.Component {
       bar: barChartProps,
     };
 
-    const panelChart = (
-      <Chart {...chartPropsForType[type]} config={this.props.config} />
```
Comment (ezr-ondrej) [inline webpack/assets/javascripts/react_app/components/ChartBox/ChartBox.js:75]: `props.config` was already destructed to `config`, using it, no other change here.

**Comment by ekohl** [comment]
I love patches that have more functionality with less code :)

File: app/views/dashboard/_status_links.html.erb [hunk 2]
```diff
@@ -1,65 +0,0 @@
-<%
```
Comment (tbrisker) [inline app/views/dashboard/_status_links.html.erb]: Why did you delete this file? it is used by another widget (status_widget)

File: app/assets/javascripts/charts.js [hunk 3]
```diff
@@ -57,9 +57,6 @@ $.fn.flot_pie = function() {
       var percent = Math.round(item.series.percent);
       return _.escape(item.series.label) + ' (' + percent + '%)';
     });
-    $(target).bind('plotclick', function(event, pos, item) {
-      search_on_click(event, item);
-    });
```
Comment (tbrisker) [inline app/assets/javascripts/charts.js]: do we still use flot pie anywhere? or did we replace all of them with the react chart? If i'm not mistaken, it is only still used in foreman-omaha - so we could deprecate it and remove a whole bunch of code (but doesn't have to be this PR).
I would however recommend checking this change doesn't break omaha plugin.

Comment (ezr-ondrej) [reply app/assets/javascripts/charts.js]: Yeah, the function is used by omaha :(
For pie chart, but still.

Comment (tbrisker) [reply app/assets/javascripts/charts.js]: so i guess we should keep it in for now and only deprecate?

Comment (ezr-ondrej) [reply app/assets/javascripts/charts.js]: or change the `flot_pie_chart` to mount the react component and deprecate that? :thinking:

File: app/assets/javascripts/charts.js [hunk 4]
```diff
@@ -320,41 +317,6 @@ function bind_hover_event(target, formater) {
   });
 }
 
-function search_on_click(event, item) {
```
Comment (ezr-ondrej) [inline app/assets/javascripts/charts.js]: This makes me :panda_face::cry: but we have to keep this function, if we want to maintain full functionality of `flot_pie_chart()`

**Comment by tbrisker** [comment]
moving back to WoC, this needs changes as mentioned before.

File: app/assets/javascripts/charts.js (hunk 3 above)
Comment (ezr-ondrej) [reply app/assets/javascripts/charts.js]: #8366 is removing flot_pie_chart completely :smirk:

File: app/assets/javascripts/charts.js (hunk 4 above)
Comment (ezr-ondrej) [reply app/assets/javascripts/charts.js]: Or do we? Not after #8366

File: app/views/dashboard/_status_links.html.erb (hunk 2 above)
Comment (ezr-ondrej) [reply app/views/dashboard/_status_links.html.erb]: Because I don't like it :( xD
(I didn't notice the other usage)

**Comment by ezr-ondrej** [comment]
This is unblocked as pie_chart and the search will be removed independently :tada:

**Comment by xprazak2** [comment]
Works for me, any additional concerns?

**Review by xprazak2 — APPROVED** [review]

**Comment by xprazak2** [comment]
Thanks!

**Comment by ezr-ondrej** [comment]
Revert of the `searchable_links` removal: #8460
